### ISSUE TITLE
Turtle-graphics extensions

### ISSUE THREAD

**Issue by davidedc** [davidedc/livecodelab#250]
Preamble:
* this is not going to change the grammar - so this will transparently work in lang v2 (modulo adding the keywords I guess, see below).
* this is going to stay un-merged in a branch until we properly like it.

I'm going to add a couple of keywords like penUp and penDown and showTurtle and hideTurtle, forward, backward, turnRight, turnLeft (both in degrees), arc.

I'm also gonna add an "axes" keyword which is gonna draw the axis according to the world matrix when "axes" is called.

This spawned out of a maddening "what the heck should be the order of the transformations here" moment, where I realised that drawing the axes would make everything so much easier to understand... and finally realising what several tweets have pointed out already, that LCL can also be used as a 3d version of Logo...

If that can be done with a few new keywords then it'd look like a win...

If the mapping is as direct as it looks at first sight then I'm gonna try to flesh it out tonight...

(Let's see how it pans out, we can scrap if we don't like.)

**Comment by rumblesan** [comment]
this sounds like it could be hilarious haha. I'm all for it

**Comment by davidedc** [comment]
I've pushed it in dedicated branch and published in secret directory here:

http://livecodelab.net/play2.1/

Some examples:
* http://livecodelab.net/play2.1/index.html#bookmark=turtleGraphicstreeDemo
* http://livecodelab.net/play2.1/index.html#bookmark=turtleGraphicsrandomGridDemo
* http://livecodelab.net/play2.1/index.html#bookmark=turtleGraphicsturtleSpiralDemo

I didn't add the penUp / penDown yet, and there are a couple of tweaks and questions...

There is an "axes" command not shown in the examples that draws the axes (with the world transformations applied when invoked)

...play around with it as you please, pass over interesting stuff or examples that you find or comments...

**Comment by davidedc** [comment]
![screen shot 2015-06-21 at 11 11 36](https://cloud.githubusercontent.com/assets/769568/8271276/89c49ca6-1806-11e5-9787-ab874163bc59.png)
![screen shot 2015-06-21 at 11 12 16](https://cloud.githubusercontent.com/assets/769568/8271278/8cbc4314-1806-11e5-80db-40afba521f21.png)

**Comment by rumblesan** [comment]
hahaha, so this is really cool. will start having a play around

**Comment by rumblesan** [comment]
just a quick one, I think that axes command would be really useful in general. especially when teaching people

**Comment by davidedc** [comment]
indeed, that's how the rabbit hole started. I needed to see the axes, and then really one can see them as the current "cursor", or "pen" or "turtle"...

**Comment by davidedc** [comment]
:turtle: :turtle: :turtle: :turtle: :turtle: :turtle: :turtle: :turtle: :turtle: :turtle: :turtle: :turtle: :turtle: :turtle: 
![screen shot 2015-06-21 at 12 49 10](https://cloud.githubusercontent.com/assets/769568/8271510/1fddd1d2-1814-11e5-85c6-6d118b08e0bb.png)

**Comment by rumblesan** [comment]
hey, so I've rebased the changes you made for the turtle onto the new master branch. There were a couple of minor things I've fixed up but it seems to be working. Though I couldn't get it to actually draw the turtle, just a red line.

**Comment by davidedc** [comment]
ops might not have pushed the "turtle" change yet.

I'm concerned with the turtle being too cutesy for LCL. So I thought maybe, in addition to showTurtle/hideTurtle, there should be a showPen/hidePen.

because of the penUp/penDown keywords people would normally show the pen (it would look like an arrow I think), but the cutesy-inspired could show the turtle instead...
